### ISSUE TITLE
make timestamps a decorator and an interface instead of a parent class

### ISSUE THREAD

**Issue by muametgrooby** [typegoose/typegoose#792]
## Describe what you need | want

Since we can't extend multiple classes in typescript with ease, timestamps class gets in the way. Can we please turn it into a decorator or an option and make it an interface instead?

## Do you have already an idea for the implementation?

*no*

**Comment by hasezoey** [comment]
> Can we please turn it into a decorator

as in a decorator that automatically extends the class? then no, this would be against the reason the class exists: provide types for the `updatedAt` and `createdAt` properties

> an option and make it an interface instead?

probably a better idea, but then what do you want to do with it? a `class T implements TimeStamps` would still be required to re-define those fields, which this class aims to not make your do (it only provides types, nothing at runtime aside from setting the schema-option)

i am also thinking about removing the default classes again in some major version

TL;DR: if you want to use the classes properties, but have to extend another class, then i recommend to just copy the fields from the [`TimeStamps`](https://github.com/typegoose/typegoose/blob/e42769504d0fe60bee0d96785b3c09cc2b1e70b0/src/defaultClasses.ts#L5-L12) class into the class you are currently in and set the schema-option manually, or have your base class extend it

**Comment by hasezoey** [comment]
closing because of no activity and "wont fix" label
